CMakeLists.txt

```diff
@@ -430,8 +430,12 @@ if (WITH_LLVM)
     endforeach()
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DNDEBUG")
 
-    llvm_map_components_to_libnames(llvm_libs_direct ${SYMENGINE_LLVM_COMPONENTS})
-    llvm_expand_dependencies(llvm_libs ${llvm_libs_direct})
+    if (LLVM_LINK_LLVM_DYLIB)
+      set(llvm_libs LLVM)
+    else()
+      llvm_map_components_to_libnames(llvm_libs_direct ${SYMENGINE_LLVM_COMPONENTS})
+      llvm_expand_dependencies(llvm_libs ${llvm_libs_direct})
+    endif()
 
```

Comment:
We need a way out of this. i.e. a way to force use static libraries even if there is a shared library

Reply:
Does this look better? I've added `WITH_LLVM_DYLIB` switch.